subsys/net/openthread/rpc/server/ot_rpc_srp_client.c

```diff
@@ -0,0 +1,863 @@
+
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+
+	nrf_rpc_cbor_rsp_no_err(group, &rsp_ctx);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_get_server_address,
+			 OT_RPC_CMD_SRP_CLIENT_GET_SERVER_ADDRESS,
+			 ot_rpc_cmd_srp_client_get_server_address, NULL);
+
+static void ot_rpc_cmd_srp_client_get_ttl(const struct nrf_rpc_group *group,
+					  struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	uint32_t ttl;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	ttl = otSrpClientGetTtl(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_uint(group, ttl);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_get_ttl,
+			 OT_RPC_CMD_SRP_CLIENT_GET_TTL, ot_rpc_cmd_srp_client_get_ttl, NULL);
+
+static void ot_rpc_cmd_srp_client_get_lease_interval(const struct nrf_rpc_group *group,
+						     struct nrf_rpc_cbor_ctx *ctx,
+						     void *handler_data)
+{
+	uint32_t lease_int;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	lease_int = otSrpClientGetLeaseInterval(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_uint(group, lease_int);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_get_lease_interval,
+			 OT_RPC_CMD_SRP_CLIENT_GET_LEASE_INTERVAL,
+			 ot_rpc_cmd_srp_client_get_lease_interval,
+			 NULL);
+
+static void ot_rpc_cmd_srp_client_get_key_lease_interval(const struct nrf_rpc_group *group,
+							 struct nrf_rpc_cbor_ctx *ctx,
+							 void *handler_data)
+{
+	uint32_t key_lease_int;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	key_lease_int = otSrpClientGetKeyLeaseInterval(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_uint(group, key_lease_int);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_get_key_lease_interval,
+			 OT_RPC_CMD_SRP_CLIENT_GET_KEY_LEASE_INTERVAL,
+			 ot_rpc_cmd_srp_client_get_key_lease_interval, NULL);
+
+
+static void ot_rpc_cmd_srp_client_set_host_addresses(const struct nrf_rpc_group *group,
+						     struct nrf_rpc_cbor_ctx *ctx,
+						     void *handler_data)
+{
+	otError error;
+	size_t index = 0;
+	uint8_t num = nrf_rpc_decode_uint(ctx);
+
+	/* Check if more memory is needed */
+	if (host_data.addresses && (num > host_data.addresses_num)) {
+		free(host_data.addresses);
+		host_data.addresses = malloc(num * OT_IP6_ADDRESS_SIZE);
+	} else if (!host_data.addresses) {
+		host_data.addresses = malloc(num * OT_IP6_ADDRESS_SIZE);
+	}
+
+	host_data.addresses_num = num;
+
+	zcbor_list_start_decode(ctx->zs);
+
+	for (; index < num; ++index) {
+		nrf_rpc_decode_buffer(ctx, &host_data.addresses[index], OT_IP6_ADDRESS_SIZE);
```

Comment:
nit: what's quite unsafe about this is that `host_data.addresses` array may already be passed to OpenThread and you start decoding new data to this array before taking a mutex. Also, the decoding may go wrong which will leave the array in a non-deterministic state. Therefore, although it may result in more allocations, it's safer to always decode new data into a new buffer that only gets used if everything went well, or to do the decoding with the mutex held (and release the buffer on error).